### PR TITLE
Using switches to restrict input for source control provider option

### PR DESCRIPTION
**Pull request by hannah-smartbear**
- Restricting the the source control provider option for the `create-build` command to only accept `github`, `github-enterprise`, `bitbucket`, `bitbucket-server`, `gitlab`, or `gitlab-onpremise`
- If the user enters an invalid provider, they will see an error and the build will not be sent

### CODE REVIEW

File: pkg/utils/source-control-options.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+func SourceControl(provider string, logger log.Logger) string {
+	if provider == "" {
+		return provider
+	}
+	switch provider {
+	case "github", "github-enterprise", "bitbucket", "bitbucket-server", "gitlab", "gitlab-onpremise":
+		return provider
+	default:
+		logger.Warn(fmt.Sprintf("%s is not an accepted value for the source control provider, proceeding without a provider", provider))
```
Comment (joshedney) [inline pkg/utils/source-control-options.go:16]: @tomlongridge Did we want to proceed without this information or did we want to fail and exit ?

Comment (tomlongridge) [reply pkg/utils/source-control-options.go:16]: Fail and exit if they've attempted to provide a value.

File: pkg/options/options.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ type CLI struct {
 		Dsym               upload.Dsym                   `cmd:"" help:"Upload dSYMs for iOS"`
 		UnityAndroid       upload.UnityAndroid           `cmd:"" help:"Upload Android mappings and NDK symbol files from Unity projects"`
 	} `cmd:"" help:"Upload symbol/mapping files"`
-	CreateBuild          CreateBuild          `cmd:"" help:"Provide extra information whenever you build, release, or deploy your application"`
+	CreateBuild          CreateBuild          `cmd:"" help:"Provide extra information whenever you build, release, or deploy your application. If providing source control information, accepted values for provider are: github, github-enterprise, bitbucket, bitbucket-server, gitlab, gitlab-onpremise"`
```
Comment (tomlongridge) [inline pkg/options/options.go:42]: We don't need this detail here IMO

```suggestion
	CreateBuild          CreateBuild          `cmd:"" help:"Provide extra information whenever you build, release, or deploy your application"`
```

**Review by tomlongridge — COMMENTED** [review]
Can you also add a changelog?

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**